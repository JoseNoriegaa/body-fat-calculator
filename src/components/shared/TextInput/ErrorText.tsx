// External dependencies
import React from 'react';

// Internal dependencies
import useStyles from './ErrorText.styles';

function ErrorText({ children }: React.PropsWithChildren<unknown>) {
  const classes = useStyles();
  return <p className={classes.error}>{children}</p>;
}

export default ErrorText;
