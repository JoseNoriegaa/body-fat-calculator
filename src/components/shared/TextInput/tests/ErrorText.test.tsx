// External dependencies
import { render, screen } from '@testing-library/react';

// Internal dependencies
import ErrorText from '../ErrorText';

describe('<ErrorText />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <ErrorText>
        <span>error text</span>
      </ErrorText>
    );

    expect(screen.getByText('error text')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
