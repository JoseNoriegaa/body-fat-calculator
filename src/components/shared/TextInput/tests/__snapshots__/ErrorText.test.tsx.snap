// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorText /> should match snapshot 1`] = `
<DocumentFragment>
  <p
    class="error-0-2-1"
  >
    <span>
      error text
    </span>
  </p>
</DocumentFragment>
`;
